cockroachdb/templates/job.init.yaml

```diff
@@ -33,14 +33,20 @@ spec:
     spec:
       restartPolicy: OnFailure
       terminationGracePeriodSeconds: 0
-    {{- if or .Values.image.credentials (and .Values.tls.enabled .Values.tls.init.image.credentials (not .Values.tls.certs.provided) (not .Values.tls.certs.certManager)) }}
+    {{- if or .Values.image.credentials .Values.image.imagePullSecrets (and .Values.tls.enabled .Values.tls.init.image.credentials (not .Values.tls.certs.provided) (not .Values.tls.certs.certManager)) (and .Values.tls.enabled .Values.tls.init.image.imagePullSecrets (not .Values.tls.certs.provided) (not .Values.tls.certs.certManager)) }}
       imagePullSecrets:
       {{- if .Values.image.credentials }}
         - name: {{ template "cockroachdb.fullname" . }}.db.registry
       {{- end }}
+      {{- if .Values.image.imagePullSecrets }}
```

Comment:
I think it might be confusing that the value is called `imagePullSecrets` (in plural form), but you can actually just specify a single secret. I would either do something like this 
```
    {{- if .Values.image.imagePullSecrets }}
      imagePullSecrets:
      {{- range .Values.image.imagePullSecrets }}
        - name: {{ . }}
      {{- end}}
    {{- end }}
```
or rename the variable to `imagePullSecret`. (I actually prefer the first solution).